### PR TITLE
Added updateByQuery endpoint

### PR DESCRIPTION
**Pull request by mmoreram**
- Added Changelog entry as well under Added block

### CODE REVIEW

**Comment by mmoreram** [comment]
Solves #1103

File: test/Elastica/IndexTest.php [hunk 1]
```diff
@@ -290,6 +291,76 @@ public function testDeleteByQueryWithQueryAndOptions()
         $this->assertEquals(0, $response->count());
     }
 
+    /**
+     * @group functional
+     * @group mmoreram
```
Comment (ruflin) [inline test/Elastica/IndexTest.php]: What does the second group here mean and what is it for?

File: test/Elastica/IndexTest.php [hunk 2]
```diff
@@ -290,6 +291,76 @@ public function testDeleteByQueryWithQueryAndOptions()
...
+        $type1->addDocument(new Document(2, ['name' => 'ruflin']));
+        $index->refresh();
+
+        $response = $index->search('ruflin*');
+        $this->assertEquals(2, $response->count());
+
+        $response = $index->search('nicolas');
+        $this->assertEquals(1, $response->count());
+
+        // Delete first document
```
Comment (ruflin) [inline test/Elastica/IndexTest.php]: This only renames it and does not delete it?

File: test/Elastica/IndexTest.php [hunk 3]
```diff
@@ -290,6 +291,76 @@ public function testDeleteByQueryWithQueryAndOptions()
...
+        $response = $index->updateByQuery('nicolas', new Script('ctx._source.name = "marc"'));
+        $this->assertTrue($response->isOk());
+
+        $index->refresh();
+
+        // Makes sure, document is deleted
```
Comment (ruflin) [inline test/Elastica/IndexTest.php]: See comment above, it makes sure document is renamed.

File: test/Elastica/IndexTest.php [hunk 4]
```diff
@@ -290,6 +291,76 @@ public function testDeleteByQueryWithQueryAndOptions()
...
+        $type1->addDocument(new Document(2, ['name' => 'ruflin']));
+        $index->refresh();
+
+        $response = $index->search('ruflin*');
+        $this->assertEquals(2, $response->count());
+
+        $response = $index->search('nicolas');
+        $this->assertEquals(1, $response->count());
+
+        // Delete first document
```
Comment (ruflin) [inline test/Elastica/IndexTest.php]: Updates all docs?

File: test/Elastica/IndexTest.php [hunk 5]
```diff
@@ -290,6 +291,76 @@ public function testDeleteByQueryWithQueryAndOptions()
...
+        $response = $index->updateByQuery('*', new Script('ctx._source.name = "marc"'));
+        $this->assertTrue($response->isOk());
+
+        $index->refresh();
+
+        // Makes sure, document is deleted
```
Comment (ruflin) [inline test/Elastica/IndexTest.php]: See above

**Comment by mmoreram** [comment]
Fixed all docs. I'm sorry, my mistake I didn't check properly tests :)

**Comment by ruflin** [comment]
@mmoreram Thanks a lot for the contribution.

**Comment by mmoreram** [comment]
@ruflin thanks to you, for this library :=)

When do you expect you will release next stable version?

**Comment by ruflin** [comment]
Here you go: http://elastica.io/2018/05/29/release-6-dot-0-2/

**Comment by mmoreram** [comment]
@ruflin WEEEE

That deserves a beer if you come to Barcelona ! :D

Thanks so much!

**Pull request merged**